### PR TITLE
Increase language clarity in challenge "Basic Javascript: Iterate Through an Array with a For Loop"

### LINKED ISSUE #36030: Phrasing Issue on Basic JavaScript: Iterate Through an Array with a For Loop
I think the below is phrased in a confusing way.

```
Remember that Arrays have zero-based numbering, which means the last index of the array is length - 1. Our condition for this loop is i < arr.length, which stops when i is at length - 1.
```

I thought it was saying "the last index of the array has a length of -1".

This was very strange to me because how could an array index have a length of negative 1?

If it was using the grey background with red lettering, and it said "arr.length-1", then it would be much more clear. 

I nearly posted on the forum "How can an index have a length of negative 1?" - because I thought maybe when you loop past an index that exists, it loops back on itself.

Anyway - it may be worth updating.

### PR DESCRIPTION
**Pull request by organicdude**
- [ x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x ] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x ] My pull request targets the `master` branch of freeCodeCamp.
- [x ] None of my changes are plagiarized from another source without proper attribution.
- [ x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [ x] My changes do not use shortened URLs or affiliate links.

Closes #36030

### CODE REVIEW

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-through-an-array-with-a-for-loop.english.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ for (var i = 0; i < arr.length; i++) {
 }
 ```
 
-Remember that Arrays have zero-based numbering, which means the last index of the array is length - 1. Our <dfn>condition</dfn> for this loop is <code>i < arr.length</code>, which stops when <code>i</code> is at length - 1.
+Remember that Arrays have zero-based numbering, which means the last index of the array is <code>length - 1</code>. Our <dfn>condition</dfn> for this loop is <code>i < arr.length</code>, which stops when <code> i </code> is at <code>length - 1</code>.
```
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-through-an-array-with-a-for-loop.english.md]: While we are fixing the wording for this challenge, let's change `numbering` to `indexing` and also remove the spaces inside the one code tag.
```suggestion
Remember that Arrays have zero-based indexing, which means the last index of the array is <code>length - 1</code>. Our <dfn>condition</dfn> for this loop is <code>i < arr.length</code>, which stops when <code>i</code> is at <code>length - 1</code>.
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
@organicdude Looks good.  I have one new suggestion and a request to fix another.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-through-an-array-with-a-for-loop.english.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ for (var i = 0; i < arr.length; i++) {
 }
 ```
 
-Remember that Arrays have zero-based numbering, which means the last index of the array is length - 1. Our <dfn>condition</dfn> for this loop is <code>i < arr.length</code>, which stops when <code>i</code> is at length - 1.
+Remember that arrays have zero-based indexing, which means the last index of the array is <code>length - 1</code>. Our <dfn>condition</dfn> for this loop is <code>i < arr.length</code>, which stops when <code>i</code> is at <code>length - 1</code>.
```
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-through-an-array-with-a-for-loop.english.md]: @organicdude I think we can make this better by changing a bit more of the wording.  I will let some other mods add their comments before you make the commit though.  What do you think?
```suggestion
Remember that arrays have zero-based indexing, which means the last index of the array is <code>length - 1</code>. Our <dfn>condition</dfn> for this loop is <code>i < arr.length</code>, which stops the loop when <code>i</code> is equal to <code>length</code>.
```

**Review by RandellDawson — COMMENTED** [review]
See my new comment.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-through-an-array-with-a-for-loop.english.md (hunk 2 above)
Comment (organicdude) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-through-an-array-with-a-for-loop.english.md]: Sounds good - just please let me know if I've fallen asleep at the wheel and am not allowing a finished commit to go through.

Here's another way to phrase it - more noob friendly IMHO.
```suggestion
Remember that arrays first index is <code>0</code>, not <code>1</code>. That means that if <code>i</code> is your array's index, then the last the last defined index will be <code>length - 1</code>. That also means that your loop should stop running when <code>i > length - 1</code>.
```

Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-through-an-array-with-a-for-loop.english.md]: We'll let you know.  @Manish-Giri @thecodingaviator @ojeytonwilliams @moT01 What do you think about my suggestion?

Comment (thecodingaviator) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-through-an-array-with-a-for-loop.english.md]: @RandellDawson How does this sound.
```suggestion
Remember that arrays have zero-based indexing, which means the last index of the array is <code>length - 1</code>. Our <dfn>condition</dfn> for this loop is <code>i < arr.length</code>, which stops the loop when <code>i</code> is equal to <code>length</code>, i.e. the last iteration is when <code>i</code> is equal to <code>length-1</code>.
```

Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-through-an-array-with-a-for-loop.english.md]: @organicdude I am leaning toward's @thecodingaviator's version.  That being said, I think just adding on that last part of @thecodingaviator's suggestion makes the last sentence just too long.  I think the example part should be a separate sentence on its own.

Comment (thecodingaviator) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-through-an-array-with-a-for-loop.english.md]: @RandellDawson Feel free to edit my suggestion or make a new one in this thread to show us what you think. I'm not sure how we would go about splitting it

Comment (ojeytonwilliams) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-through-an-array-with-a-for-loop.english.md]:
```suggestion
Remember that arrays have zero-based indexing, which means the last index of the array is <code>length - 1</code>. Our <dfn>condition</dfn> for this loop is <code>i < arr.length</code>, which stops the loop when <code>i</code> is equal to <code>length</code>.  In this case the last iteration has <code>i === 4</code> and outputs <code>6</code> to the console.
```
How about this way of splitting up @thecodingaviator's version?  I thought mentioning what the loop does in the final iteration might help clear up any lingering confusion about indices.
@organicdude Arrays are introduced as [being zero-based](https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/basic-javascript/access-array-data-with-indexes/) so I think it's fine to repeat that here.

Comment (thecodingaviator) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-through-an-array-with-a-for-loop.english.md]: Building on your suggestion, I came up with this:
```suggestion
Remember that arrays have zero-based indexing, which means the last index of the array is <code>length - 1</code>. Our <dfn>condition</dfn> for this loop is <code>i < arr.length</code>, which stops the loop when <code>i</code> is equal to <code>length</code>.  In this case the last iteration is <code>i === 4</code> i.e. when <code>i</code> becomes equal to <code>arr.length</code> and outputs <code>6</code> to the console.
```

**Comment by RandellDawson** [comment]
Just to summarize, we now have the following 4 suggestions:

**Version No. 1 (my original suggestion)**
>Remember that arrays have zero-based indexing, which means the last index of the array is <code>length - 1</code>. Our <dfn>condition</dfn> for this loop is <code>i < arr.length</code>, which stops the loop when <code>i</code> is equal to <code>length</code>.

**Version No. 2 (the OP's counter suggestion)**
>Remember that an array's first index is <code>0</code>, not <code>1</code>. That means that if <code>i</code> is your array's index, then the last the last defined index will be <code>length - 1</code>. That also means that your for loop should stop running when <code>i > length - 1</code>.

**Version No. 3 (@ojeytonwilliams' suggestion)**
>Remember that arrays have zero-based indexing, which means the last index of the array is <code>length - 1</code>. Our <dfn>condition</dfn> for this loop is <code>i < arr.length</code>, which stops the loop when <code>i</code> is equal to <code>length</code>.  In this case the last iteration has <code>i === 4</code> and outputs <code>6</code> to the console.

**Version No. 4 (@thecodingaviator's suggestion)**
>Remember that arrays have zero-based indexing, which means the last index of the array is <code>length - 1</code>. Our <dfn>condition</dfn> for this loop is <code>i < arr.length</code>, which stops the loop when <code>i</code> is equal to <code>length</code>.  In this case the last iteration is <code>i === 4</code> i.e. when <code>i</code> becomes equal to <code>arr.length</code> and outputs <code>6</code> to the console.

It is time to pick one and not accept any more suggestions.  I am fine with version 3 or 4.

**Comment by ojeytonwilliams** [comment]
@RandellDawson I'm fine with @thecodingaviator's last suggestion (version 4).

**Comment by moT01** [comment]
I made the suggested changes here. I also removed the strings we don't use in the tests and a few empty lines to clean up the file.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-through-an-array-with-a-for-loop.english.md [hunk 3]
```diff
@@ -76,13 +72,11 @@ var myArr = [ 2, 3, 4, 5, 6];
 ```
 
 </div>
-
```
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-through-an-array-with-a-for-loop.english.md]: @moT01 This extra line will need to be in there.  See my PR https://github.com/freeCodeCamp/freeCodeCamp/pull/36278
```suggestion

```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
@moT01 See my requested change.

**Review by RandellDawson — APPROVED** [review]
LGTM

**Review by thecodingaviator — APPROVED** [review]
LGTM

**Comment by thecodingaviator** [comment]
Thanks for contributing to the freeCodeCamp repository. Congratulations on your first PR! 🎊 

Look forward to more contributions from you! ✈️
